### PR TITLE
CTCTRADERS-505: Update Notification writes to create UserAnswers

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/models/messages/EventDetails.scala [hunk 1]
```diff
@@ -40,24 +40,9 @@ object EventDetails {
...
-
-    Transhipment.reads or
-      Incident.format
-  }
-
-  implicit lazy val writes: Writes[EventDetails] = Writes {
-    case i: Incident     => Json.toJson(i)(Incident.format)
-    case t: Transhipment => Json.toJson(t)(Transhipment.writes)
+  implicit lazy val writes: OWrites[EventDetails] = OWrites {
+    case i: Incident     => Json.toJsObject(i)(Incident.format)
```
Comment (ashwinirudrappa) [inline app/models/messages/EventDetails.scala:44]: @MatthewAlexanderBeck No need to have format here I guess

File: app/models/messages/EventDetails.scala [hunk 2]
```diff
@@ -139,24 +124,9 @@ object Transhipment {
...
-    VehicularTranshipment.reads or
-      ContainerTranshipment.format
-  }
-
-  implicit lazy val writes: Writes[Transhipment] = Writes {
-    case t: VehicularTranshipment => Json.toJson(t)(VehicularTranshipment.writes)
-    case t: ContainerTranshipment => Json.toJson(t)(ContainerTranshipment.format)
+  implicit lazy val writes: OWrites[Transhipment] = OWrites {
+    case t: VehicularTranshipment => Json.toJsObject(t)(VehicularTranshipment.writes)
+    case t: ContainerTranshipment => Json.toJsObject(t)(ContainerTranshipment.format)
```
Comment (ashwinirudrappa) [inline app/models/messages/EventDetails.scala:129]: @MatthewAlexanderBeck Better to have writes alone

File: app/models/messages/EventDetails.scala [hunk 3]
```diff
@@ -181,34 +151,20 @@ object VehicularTranshipment {
...
-            "transportIdentity" -> transhipment.transportIdentity,
-            "transportCountry"  -> transhipment.transportCountry,
-            "date"              -> transhipment.date,
-            "authority"         -> transhipment.authority,
-            "place"             -> transhipment.place,
-            "country"           -> transhipment.country,
-            "containers"        -> Json.toJson(transhipment.containers)
+            "transportIdentity"    -> transhipment.transportIdentity,
+            "transportNationality" -> Json.obj("state" -> "", "code" -> transhipment.transportCountry, "description" -> ""),
+            "date"                 -> transhipment.date,
```
Comment (ashwinirudrappa) [inline app/models/messages/EventDetails.scala:161]: @MatthewAlexanderBeck We don't use them, do we need to have it in code

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 4]
```diff
@@ -20,6 +20,7 @@ import java.time.LocalDate
 
 import derivable.DeriveNumberOfEvents
 import models.messages._
+import models.reference.Country
```
Comment (ashwinirudrappa) [inline app/services/conversion/ArrivalNotificationConversionService.scala:23]: @MatthewAlexanderBeck It's an unused import

File: test/generators/MessagesModelGenerators.scala [hunk 5]
```diff
@@ -21,6 +21,7 @@ import java.time.{LocalDate, LocalTime}
 import models._
 import models.messages.ErrorType.{GenericError, MRNError}
 import models.messages._
+import models.reference.Country
```
Comment (ashwinirudrappa) [inline test/generators/MessagesModelGenerators.scala:24]: Unused import

File: app/models/messages/EventDetails.scala (hunk 2 above)
Comment (MatthewAlexanderBeck) [reply app/models/messages/EventDetails.scala:129]: @ashwinirudrappa we need this writes for the EnRouteEvents model as it takes in a generic type

File: app/services/conversion/UserAnswersConversionService.scala [hunk 6]
```diff
@@ -16,96 +16,16 @@
...
-          ua6 <- ua5
-            .set(GoodsLocationPage, BorderForceOffice)
-          ua7 <- ua6
-            .set(IncidentOnRoutePage, normalNotification.enRouteEvents.isDefined)
-        } yield
-          enRouteEvents(normalNotification, userAnswers) match {
-            case Some(eventObj) => ua7.copy(data = ua7.data ++ eventObj.data)
-            case _              => ua7
-          }).toOption
+        Some(UserAnswers(normalNotification.movementReferenceNumber, Json.toJsObject(normalNotification)))
```
Comment (amaalali) [inline app/services/conversion/UserAnswersConversionService.scala:28]: So at the moment we only support the Normal Notification?

File: test/models/messages/EnRouteEventSpec.scala [hunk 7]
```diff
@@ -247,10 +240,14 @@ class EnRouteEventSpec
...
+      "eventReported" -> enRouteEvent.alreadyInNcts,
+      "eventCountry"  -> Json.obj("state" -> "", "code" -> enRouteEvent.countryCode, "description" -> ""),
       "seals"         -> Json.toJson(enRouteEvent.seals)
-    )
+    ) ++ enRouteEvent.eventDetails
+      .map {
+        result =>
+          Json.toJsObject(result).filterNulls
+      }
+      .getOrElse(JsObject.empty)
```
Comment (amaalali) [inline test/models/messages/EnRouteEventSpec.scala:252]: This looks suspiciously like the implementation

**Pull request merged**